eos/scattering/ee-to-ccbar.hh

```diff
@@ -0,0 +1,405 @@
+    8   D0   D*0bar   VP           3       -
+    9   D*0  D0bar    VP           3       8 (c.c.)
+    10  D+   D*-      VP           3       8 (isospin)
+    11  D*+  D-       VP           3       8 (c.c.)
+    12  Ds   Ds       PP (P)       3       -
+    13  D*0  D*0bar   VV (P, S=0)  3       -
+    14  D*0  D*0bar   VV (P, S=2)  3       -
+    15  D*0  D*0bar   VV (F, S=2)  7       -
+    16  D*+  D*-      VV (P, S=0)  3       13 (isospin)
+    17  D*+  D*-      VV (P, S=2)  3       14 (isospin)
+    18  D*+  D*-      VV (F, S=2)  7       15 (isospin)
+    19  Ds+  Ds*-     VP           3       -
+    20  Ds*+ Ds-      VP           3       19 (c.c.)
+    21  Ds*+ Ds*-     VV (P, S=0)  3       -
+    22  Ds*+ Ds*-     VV (P, S=2)  3       -
+    23  Ds*+ Ds*-     VV (F, S=2)  7       -
+    */
+
+    // S -> PP channel
+    template <unsigned nchannels_, unsigned nresonances_, unsigned order_>
+    struct SPPchan :
+    public KMatrix<nchannels_, nresonances_, order_>::Channel
+    {
+
+        SPPchan(std::string name, double m1, double m2, unsigned l_orbital, std::array<Parameter, nresonances_> g0s) :
+            KMatrix<nchannels_, nresonances_, order_>::Channel(name, m1, m2, l_orbital, g0s)
+        {
+        };
+
+        //Usefull definitions for beta and rho
+        const double mm = this->_m1 - this->_m2;
+        const double mp = this->_m1 + this->_m2;
+        //sqrt of the Källen factor, defined with an absolute value
+        double sqlk(const double & s) {
+            return std::sqrt(std::abs((s-mp*mp)*(s-mm*mm)));
+        }
+        const double pi = M_PI;
+        const complex<double> i =  complex<double>(0.0, 1.0);
+
+        double beta(const double & s){
+            if (s < mp*mp) { return 0.; } //Kinematic threshold
+            else { return sqlk(s)/s; }
+        }
+
+        complex<double> rho(const double & s) {
+            complex<double> result = 0.0;
+            if (s < mm*mm){
+                // result += mm*(mp*mp-s)*std::log((mm+mp)/(mp-mm));
+                // result += mp*sqlk(s)*std::log((mm*mm+mp*mp-2*s-2*sqlk(s))/(mp*mp-mm*mm));
+                // result *= i/(mp*pi*s);
+                return result;
+            }
+            else if (s < mp*mp){
+                // result += mm*(mp*mp-s)*std::log((mm+mp)/(mp-mm));
+                // result += 2*mp*sqlk(s)*std::atan(std::sqrt((s-mm*mm)/(mp*mp-s)));
+                // result *= i/(mp*pi*s);
+                return result;
+            }
+            else {
+                // result += mm*(mp*mp-s)*std::log((mm+mp)/(mp-mm));
+                // result += mp*sqlk(s)*std::log((2*s+2*sqlk(s)-mm*mm-mp*mp)/(mp*mp-mm*mm));
+                // result *= i/(mp*pi*s);
+                result += sqlk(s)/s;
+                return result;
+            }
+        }
+    };
+
+
+    // V -> PP channel
+    template <unsigned nchannels_, unsigned nresonances_, unsigned order_>
+    struct PP_Pwavechan :
```

Comment:
You use a different name than for ``SPP`` above. Please make this consistent.
Also, please adhere to coding style: class names are ``CamelCaseNoUnderscores``.

Reply:
I remove `SPP` entirely, it was not used